lib/resourceutil/work.go

```diff
@@ -15,30 +15,31 @@ import (
 // labels[WorkPipelineNameLabel] = pipelineName
 // labels[WorkTypeLabel] = v1alpha1.WorkTypeResource
 
-func SetPromiseWorkLabels(l map[string]string, promiseName, pipelineName string) {
-	setWorkLabels(l, promiseName, "", pipelineName)
+func SetPromiseWorkLabels(l map[string]string, promiseName, pipelineName, workflowType string) {
+	setWorkLabels(l, promiseName, "", pipelineName, workflowType)
```

Comment:
```suggestion
	setWorkLabels(l, promiseName, "", pipelineName, v1alpha1.WorkTypePromise)
```
It would be better to call this function with `v1alpha1.WorkTypePromise` directly instead of calling with function parameter. The function parameter passing here is not the work type, but the workflow type.